src/golang/lib/airflow/client.go

```diff
@@ -39,15 +39,34 @@ func newClient(ctx context.Context, authConf auth.Config) (*client, error) {
 
 // getDagRuns returns all of the Airflow DAGRuns for the Airflow DAG specified.
 func (c *client) getDagRuns(dagId string) ([]airflow.DAGRun, error) {
-	dagRunsResp, resp, err := c.apiClient.DAGRunApi.GetDagRuns(
-		c.ctx,
-		dagId,
-	).OrderBy("start_date").Execute()
-	if err != nil {
-		return nil, wrapApiError(err, "GetDagRuns", resp)
+	limit := 100 // This is the max number of DAG runs that can be returned in each response.
+	offset := 0
+	var dagRuns []airflow.DAGRun
+
+	// Keep paginating through DAG runs until there are none in response
+	for {
+		dagRunsResp, resp, err := c.apiClient.DAGRunApi.GetDagRuns(
+			c.ctx,
+			dagId,
+		).
+			OrderBy("start_date").
+			Limit(int32(limit)).
+			Offset(int32(offset)).
+			Execute()
+		if err != nil {
+			return nil, wrapApiError(err, "GetDagRuns", resp)
+		}
+
+		if len(*dagRunsResp.DagRuns) == 0 {
+			// There are no more DAG Runs
+			break
+		}
+
+		dagRuns = append(dagRuns, *dagRunsResp.DagRuns...)
+		offset += limit
```

Comment:
should this be `len(*dagRunsResp.DagRuns)`? I'm curious if it only has say 57 runs and the next API call will try to fetch with offset 100, will this cause any issue?

Reply:
Since we are doing this in order, if there are less than 100 runs returned, this means there are only less than 100 runs returned. The next fetch will return 0 runs.

I tested this with a DAG that had 9 runs. The first fetch returned 9 and the next one returned 0.